### PR TITLE
Fix obsolete code in `ParameterizedInvocationNameFormatter`

### PR DESCRIPTION
**Pull request by Pankraz76**
while inspecting 

-https://github.com/junit-team/junit-framework/issues/4714

i was wondering whats all these obsolete code used for?

---

I hereby agree to the terms of the [JUnit Contributor License Agreement](https://github.com/junit-team/junit-framework/blob/002a0052926ddee57cf90580fa49bc37e5a72427/CONTRIBUTING.md#junit-contributor-license-agreement).

---

### Definition of Done

- [ ] There are no TODOs left in the code
- [ ] Method [preconditions](https://docs.junit.org/snapshot/api/org.junit.platform.commons/org/junit/platform/commons/util/Preconditions.html) are checked and documented in the method's Javadoc
- [ ] [Coding conventions](https://github.com/junit-team/junit-framework/blob/HEAD/CONTRIBUTING.md#coding-conventions) (e.g. for logging) have been followed
- [ ] Change is covered by [automated tests](https://github.com/junit-team/junit-framework/blob/HEAD/CONTRIBUTING.md#tests) including corner cases, errors, and exception handling
- [ ] Public API has [Javadoc](https://github.com/junit-team/junit-framework/blob/HEAD/CONTRIBUTING.md#javadoc) and [`@API` annotations](https://apiguardian-team.github.io/apiguardian/docs/current/api/org/apiguardian/api/API.html)
- [ ] Change is documented in the [User Guide](https://docs.junit.org/snapshot/user-guide/) and [Release Notes](https://docs.junit.org/snapshot/user-guide/#release-notes)

### CODE REVIEW

File: junit-jupiter-params/src/main/java/org/junit/jupiter/params/ParameterizedInvocationNameFormatter.java [hunk 1]
```diff
@@ -247,22 +240,8 @@ public synchronized void append(ArgumentsContext context, StringBuffer result) {
 		}
 
 		private @Nullable Object[] makeReadable(@Nullable Object[] arguments) {
-			Format[] formats = messageFormat.getFormatsByArgumentIndex();
-			@Nullable
-			Object[] result = Arrays.copyOf(arguments, Math.min(arguments.length, formats.length), Object[].class);
-			for (int i = 0; i < result.length; i++) {
-				if (formats[i] == null) {
```
Comment (Pankraz76) [inline junit-jupiter-params/src/main/java/org/junit/jupiter/params/ParameterizedInvocationNameFormatter.java:254]: Condition 'formats[i] == null' is always 'false'

Comment (Pankraz76) [reply junit-jupiter-params/src/main/java/org/junit/jupiter/params/ParameterizedInvocationNameFormatter.java:254]: and after this the whole code collapses.

File: junit-jupiter-params/src/main/java/org/junit/jupiter/params/ParameterizedInvocationNameFormatter.java [hunk 2]
```diff
@@ -229,15 +227,10 @@ public void append(ArgumentsContext context, StringBuffer result) {
 
 	private static class MessageFormatPartialFormatter implements PartialFormatter {
 
-		@SuppressWarnings("UnnecessaryUnicodeEscape")
-		private static final char ELLIPSIS = '\u2026';
-
 		private final MessageFormat messageFormat;
-		private final int argumentMaxLength;
 
-		MessageFormatPartialFormatter(String pattern, int argumentMaxLength) {
```
Comment (Pankraz76) [inline junit-jupiter-params/src/main/java/org/junit/jupiter/params/ParameterizedInvocationNameFormatter.java:238]: API seems questionable then.

**Comment by sormuras** [comment]
Related to https://github.com/junit-team/junit-framework/commit/bff769dcc6580f6f74ed3a32f70999629363ba4d

File: junit-jupiter-params/src/main/java/org/junit/jupiter/params/ParameterizedInvocationNameFormatter.java [hunk 3]
```diff
@@ -247,22 +240,8 @@ public synchronized void append(ArgumentsContext context, StringBuffer result) {
...
-			}
-			return result;
-		}
-
-		private @Nullable String truncateIfExceedsMaxLength(@Nullable String argument) {
-			if (argument != null && argument.length() > this.argumentMaxLength) {
-				return argument.substring(0, this.argumentMaxLength - 1) + ELLIPSIS;
-			}
-			return argument;
+			return Arrays.copyOf(arguments,
```
Comment (Pankraz76) [inline junit-jupiter-params/src/main/java/org/junit/jupiter/params/ParameterizedInvocationNameFormatter.java:243]:
```suggestion
			return replaceNonPrintableCharacters(Arrays.copyOf(arguments,
```
you suggest like this, right?
@sbrannen

i would like to impl. this issue. Seems possible with limited scope.

**Comment by marcphilipp** [comment]
This code isn't obsolete at all. The array type should be `@Nullable Format[]` which I've rectified in 7ea357d08d577c235a311f1debf89ca194c7fc33.

@Pankraz76 Before sending a PR, please run at least the unit test for the code you're changing.

**Comment by sormuras** [comment]
> Condition 'formats[i] == null' is always 'false'

Seems like your and IDEA's assumption is wrong.
image

An element of the array can be `null`. See [MessageFormat.html#getFormatsByArgumentIndex()](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/text/MessageFormat.html#getFormatsByArgumentIndex()) for details:
> If an argument index is not used for any format element in the pattern string, then null is returned in the array.

**Comment by Pankraz76** [comment]
> @Pankraz76 Before sending a PR, please run at least the unit test for the code you're changing.

sorry what a shame, my bad.

the build is not working on my system im stuck here in main too:

```
WARNING: A terminally deprecated method in sun.misc.Unsafe has been called
WARNING: sun.misc.Unsafe::objectFieldOffset has been called by com.google.common.util.concurrent.AbstractFuture$UnsafeAtomicHelper (file:/Users/vincent.potucek/.gradle/caches/modules-2/files-2.1/com.google.guava/guava/33.4.0-jre/3fcc0a259f724c7de54a6a55ea7e26d3d5c0cac/guava-33.4.0-jre.jar)
WARNING: Please consider reporting this to the maintainers of class com.google.common.util.concurrent.AbstractFuture$UnsafeAtomicHelper
WARNING: sun.misc.Unsafe::objectFieldOffset will be removed in a future release
WARNING: A terminally deprecated method in sun.misc.Unsafe has been called
WARNING: sun.misc.Unsafe::objectFieldOffset has been called by com.google.common.util.concurrent.AbstractFuture$UnsafeAtomicHelper (file:/Users/vincent.potucek/.gradle/caches/modules-2/files-2.1/com.google.guava/guava/33.4.0-jre/3fcc0a259f724c7de54a6a55ea7e26d3d5c0cac/guava-33.4.0-jre.jar)
WARNING: Please consider reporting this to the maintainers of class com.google.common.util.concurrent.AbstractFuture$UnsafeAtomicHelper
WARNING: sun.misc.Unsafe::objectFieldOffset will be removed in a future release

> Task :jupiter-tests:test_4_12
Predictive Test Selection: 1 of 16 test classes selected with profile 'Standard' (saving 0.367s serial time)

> Task :junit-vintage-engine:testWithoutJUnit4
Predictive Test Selection: 1 of 1 test class selected with profile 'Standard' (saving 0s serial time)

> Task :platform-tests:testWoodstox
Predictive Test Selection: 1 of 5 test classes selected with profile 'Standard' (saving 2.386s serial time)

> Task :platform-tooling-support-tests:generateOpenTestHtmlReport FAILED
[Fatal Error] open-test-report.xml:268:1006: XML document structures must start and end within the same entity.
org.xml.sax.SAXParseException; systemId: file:/Users/vincent.potucek/IdeaProjects/junit-framework/platform-tooling-support-tests/build/test-results/test/junit-8528997169361745006/open-test-report.xml; lineNumber: 268; columnNumber: 1006; XML document structures must start and end within the same entity.
        at java.xml/com.sun.org.apache.xerces.internal.parsers.DOMParser.parse(DOMParser.java:262)
        at java.xml/com.sun.org.apache.xerces.internal.jaxp.DocumentBuilderImpl.parse(DocumentBuilderImpl.java:342)
        at java.xml/javax.xml.parsers.DocumentBuilder.parse(DocumentBuilder.java:206)
        at org.opentest4j.reporting.tooling.core@0.2.4/org.opentest4j.reporting.tooling.core.htmlreport.DefaultHtmlReportWriter.parseDom(DefaultHtmlReportWriter.java:269)
        at org.opentest4j.reporting.tooling.core@0.2.4/org.opentest4j.reporting.tooling.core.htmlreport.DefaultHtmlReportWriter.extractData(DefaultHtmlReportWriter.java:115)
        at org.opentest4j.reporting.tooling.core@0.2.4/org.opentest4j.reporting.tooling.core.htmlreport.DefaultHtmlReportWriter.writeHtmlReport(DefaultHtmlReportWriter.java:92)
        at org.opentest4j.reporting.cli@0.2.4/org.opentest4j.reporting.cli.HtmlReportCommand.call(HtmlReportCommand.java:69)
        at org.opentest4j.reporting.cli@0.2.4/org.opentest4j.reporting.cli.HtmlReportCommand.call(HtmlReportCommand.java:39)
        at info.picocli@4.7.7/picocli.CommandLine.executeUserObject(CommandLine.java:2031)
        at info.picocli@4.7.7/picocli.CommandLine.access$1500(CommandLine.java:148)
        at info.picocli@4.7.7/picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2469)
        at info.picocli@4.7.7/picocli.CommandLine$RunLast.handle(CommandLine.java:2461)
        at info.picocli@4.7.7/picocli.CommandLine$RunLast.handle(CommandLine.java:2423)
        at info.picocli@4.7.7/picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2277)
        at info.picocli@4.7.7/picocli.CommandLine$RunLast.execute(CommandLine.java:2425)
        at info.picocli@4.7.7/picocli.CommandLine.execute(CommandLine.java:2174)
        at org.opentest4j.reporting.cli@0.2.4/org.opentest4j.reporting.cli.ReportingCli.main(ReportingCli.java:52)

> Task :junit-vintage-engine:test_4_12
Predictive Test Selection: 1 of 17 test classes selected with profile 'Standard' (saving 13.601s serial time)

> Task :platform-tests:generateOpenTestHtmlReport FAILED
[Fatal Error] open-test-report.xml:2572:757: XML document structures must start and end within the same entity.
org.xml.sax.SAXParseException; systemId: file:/Users/vincent.potucek/IdeaProjects/junit-framework/platform-tests/build/test-results/test/junit-5741597695190640528/open-test-report.xml; lineNumber: 2572; columnNumber: 757; XML document structures must start and end within the same entity.
        at java.xml/com.sun.org.apache.xerces.internal.parsers.DOMParser.parse(DOMParser.java:262)
        at java.xml/com.sun.org.apache.xerces.internal.jaxp.DocumentBuilderImpl.parse(DocumentBuilderImpl.java:342)
        at java.xml/javax.xml.parsers.DocumentBuilder.parse(DocumentBuilder.java:206)
        at org.opentest4j.reporting.tooling.core@0.2.4/org.opentest4j.reporting.tooling.core.htmlreport.DefaultHtmlReportWriter.parseDom(DefaultHtmlReportWriter.java:269)
        at org.opentest4j.reporting.tooling.core@0.2.4/org.opentest4j.reporting.tooling.core.htmlreport.DefaultHtmlReportWriter.extractData(DefaultHtmlReportWriter.java:115)
        at org.opentest4j.reporting.tooling.core@0.2.4/org.opentest4j.reporting.tooling.core.htmlreport.DefaultHtmlReportWriter.writeHtmlReport(DefaultHtmlReportWriter.java:92)
        at org.opentest4j.reporting.cli@0.2.4/org.opentest4j.reporting.cli.HtmlReportCommand.call(HtmlReportCommand.java:69)
        at org.opentest4j.reporting.cli@0.2.4/org.opentest4j.reporting.cli.HtmlReportCommand.call(HtmlReportCommand.java:39)
        at info.picocli@4.7.7/picocli.CommandLine.executeUserObject(CommandLine.java:2031)
        at info.picocli@4.7.7/picocli.CommandLine.access$1500(CommandLine.java:148)
        at info.picocli@4.7.7/picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2469)
        at info.picocli@4.7.7/picocli.CommandLine$RunLast.handle(CommandLine.java:2461)
        at info.picocli@4.7.7/picocli.CommandLine$RunLast.handle(CommandLine.java:2423)
        at info.picocli@4.7.7/picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2277)
        at info.picocli@4.7.7/picocli.CommandLine$RunLast.execute(CommandLine.java:2425)
        at info.picocli@4.7.7/picocli.CommandLine.execute(CommandLine.java:2174)
        at org.opentest4j.reporting.cli@0.2.4/org.opentest4j.reporting.cli.ReportingCli.main(ReportingCli.java:52)

[Incubating] Problems report is available at: file:///Users/vincent.potucek/IdeaProjects/junit-framework/build/reports/problems/problems-report.html

FAILURE: Build completed with 2 failures.

1: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':platform-tooling-support-tests:generateOpenTestHtmlReport'.
> Process 'command '/Users/vincent.potucek/.gradle/jdks/bellsoft-24-aarch64-os_x.2/bin/java'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Get more help at https://help.gradle.org.
==============================================================================

2: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':platform-tests:generateOpenTestHtmlReport'.
> Process 'command '/Users/vincent.potucek/.gradle/jdks/bellsoft-24-aarch64-os_x.2/bin/java'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Get more help at https://help.gradle.org.
==============================================================================

Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/8.14.2/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 26s
443 actionable tasks: 58 executed, 385 up-to-date
➜  junit-framework git:(main) 
```

**Comment by marcphilipp** [comment]
@Pankraz76 Could you please share the `/Users/vincent.potucek/IdeaProjects/junit-framework/platform-tooling-support-tests/build/test-results/test/junit-8528997169361745006/open-test-report.xml` file? Have you tried `./gradlew clean`?

**Comment by Pankraz76** [comment]
unsure if i did gradle clean before, now this seems to be fixed running into the next issue:

Maybe i dont have a prerequisite running like graal or whatever its using:

```
> Task :jupiter-tests:test
Predictive Test Selection: 12 of 214 test classes selected with profile 'Standard' (saving 27.729s serial time)

35 tests completed, 1 failed

> Task :platform-tooling-support-tests:test FAILED

GraalVmStarterTests > runsTestsInNativeImage(Path, OutputFiles) FAILED
    org.opentest4j.AssertionFailedError: expected: <0> but was: <1>
        at app//org.junit.jupiter.api.AssertionFailureBuilder.build(AssertionFailureBuilder.java:158)
        at app//org.junit.jupiter.api.AssertionFailureBuilder.buildAndThrow(AssertionFailureBuilder.java:139)
        at app//org.junit.jupiter.api.AssertEquals.failNotEqual(AssertEquals.java:201)
        at app//org.junit.jupiter.api.AssertEquals.assertEquals(AssertEquals.java:152)
        at app//org.junit.jupiter.api.AssertEquals.assertEquals(AssertEquals.java:147)
        at app//org.junit.jupiter.api.Assertions.assertEquals(Assertions.java:541)
        at app//platform.tooling.support.tests.GraalVmStarterTests.runsTestsInNativeImage(GraalVmStarterTests.java:56)

[Incubating] Problems report is available at: file:///Users/vincent.potucek/IdeaProjects/junit-framework/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':platform-tooling-support-tests:test'.
> There were failing tests. See the report at: file:///Users/vincent.potucek/IdeaProjects/junit-framework/platform-tooling-support-tests/build/reports/tests/test/index.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/8.14.2/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 25s
503 actionable tasks: 80 executed, 1 from cache, 422 up-to-date
➜  junit-framework git:(main) 
```



[open-test-report.xml.zip](https://github.com/user-attachments/files/21143679/open-test-report.xml.zip)

**Comment by marcphilipp** [comment]
Is that reproducible? Could you please share the `platform-tooling-support-tests/build/reports/open-test-report.html` file from that run?

**Comment by Pankraz76** [comment]
[open-test-report.html.zip](https://github.com/user-attachments/files/21144385/open-test-report.html.zip)


> Is that reproducible?

on my m2 macbook air, yes.

**Comment by Pankraz76** [comment]
same for 10 y old mac:

```
Type-safe project accessors is an incubating feature.
<======-<=========----> 71% EXECUTING [5s]

> Task :platform-tooling-support-tests:test
Predictive Test Selection: 3 of 22 test classes selected with profile 'Standard' (saving 3m 58s serial time)

GraalVmStarterTests > runsTestsInNativeImage(Path, OutputFiles) FAILED
    org.opentest4j.AssertionFailedError: expected: <0> but was: <1>
        at app//org.junit.jupiter.api.AssertionFailureBuilder.build(AssertionFailureBuilder.java:158)
        at app//org.junit.jupiter.api.AssertionFailureBuilder.buildAndThrow(AssertionFailureBuilder.java:139)
        at app//org.junit.jupiter.api.AssertEquals.failNotEqual(AssertEquals.java:201)
        at app//org.junit.jupiter.api.AssertEquals.assertEquals(AssertEquals.java:152)
        at app//org.junit.jupiter.api.AssertEquals.assertEquals(AssertEquals.java:147)
        at app//org.junit.jupiter.api.Assertions.assertEquals(Assertions.java:541)
        at app//platform.tooling.support.tests.GraalVmStarterTests.runsTestsInNativeImage(GraalVmStarterTests.java:56)


35 tests completed, 1 failed

> Task :platform-tooling-support-tests:test FAILED

[Incubating] Problems report is available at: file:///Users/vincent.potucek/IdeaProjects/junit-framework/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':platform-tooling-support-tests:test'.
> There were failing tests. See the report at: file:///Users/vincent.potucek/IdeaProjects/junit-framework/platform-tooling-support-tests/build/reports/tests/test/index.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/8.14.3/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 42s
503 actionable tasks: 72 executed, 431 up-to-date
➜  junit-framework git:(main)   echo %JAVA_HOME
➜  junit-framework git:(main) uname -a
Darwin localhost 22.6.0 Darwin Kernel Version 22.6.0: Thu Apr 24 20:25:14 PDT 2025; root:xnu-8796.141.3.712.2~1/RELEASE_X86_64 x86_64
➜  junit-framework git:(main) 
```

**Comment by marcphilipp** [comment]
Thanks! Could you please try the branch from [this PR](https://github.com/junit-team/junit-framework/pull/4743)?

**Comment by Pankraz76** [comment]
now its working, thanks. (running 6 mins now, not did so before, assuming its fixed)

Was this mac related, does CI not execute all platforms?

Assuming most dev´s using mac + intellij stack.

**Comment by marcphilipp** [comment]
You had a JDK distribution installed that provided some version of the `native-image` binary but the GraalVM integration test only worked with the one from the GraalVM distribution. It worked on CI because there was no such JDK installed.

**Comment by Pankraz76** [comment]
thx
